backend/src/schemas/user.schema.ts

```diff
@@ -72,3 +73,11 @@ UserSchema.pre('save', function (next) {
   this.password = hashSync(this.password, 10);
   next();
 });
+
+UserSchema.pre('findOneAndUpdate', function (next) {
+  const update = this.getUpdate() as UpdateQuery<User>;
+  if (update && typeof update.password == 'string') {
```

Comment:
I think it should be:
```
update && update.password && typeof update.password == 'string'
```
since we want to hash when **password** is changed (what if updating user other properties beside password - then the old password would be hashed again i think)